examples_tests/44.LevelCurveExtraction/main.cpp

```diff
@@ -0,0 +1,235 @@
+	params.WindowSize = dimension2d<uint32_t>(1280, 720);
+	params.Fullscreen = false;
+	params.Vsync = true; //! If supported by target platform
+	params.Doublebuffer = true;
+	params.Stencilbuffer = false; //! This will not even be a choice soon
+	auto device = createDeviceEx(params);
+
+	if (!device)
+		return 1; // could not create selected driver.
+
+
+	//! disable mouse cursor, since camera will force it to the middle
+	//! and we don't want a jittery cursor in the middle distracting us
+	device->getCursorControl()->setVisible(false);
+
+	//! Since our cursor will be enslaved, there will be no way to close the window
+	//! So we listen for the "Q" key being pressed and exit the application
+	QToQuitEventReceiver receiver;
+	device->setEventReceiver(&receiver);
+
+
+	auto* driver = device->getVideoDriver();
+	auto* smgr = device->getSceneManager();
+    auto* am = device->getAssetManager();
+    auto* fs = am->getFileSystem();
+
+    //
+    auto* qnc = am->getMeshManipulator()->getQuantNormalCache();
+    //loading cache from file
+    qnc->loadNormalQuantCacheFromFile<asset::CQuantNormalCache::E_CACHE_TYPE::ECT_2_10_10_10>(fs,"../../tmp/normalCache101010.sse", true);
+
+    // register the zip
+    device->getFileSystem()->addFileArchive("../../media/sphere.zip");
+
+    asset::IAssetLoader::SAssetLoadParams lp;
+    auto meshes_bundle = am->getAsset("sphere.obj", lp);
+    assert(!meshes_bundle.isEmpty());
+    auto mesh = meshes_bundle.getContents().begin()[0];
+    auto mesh_raw = static_cast<asset::ICPUMesh*>(mesh.get());
+
+    //saving cache to file
+    qnc->saveCacheToFile(asset::CQuantNormalCache::E_CACHE_TYPE::ECT_2_10_10_10,fs,"../../tmp/normalCache101010.sse");
+    
+    //copy the pipeline
+    auto pipeline_cp = core::smart_refctd_ptr_static_cast<asset::ICPURenderpassIndependentPipeline>(mesh_raw->getMeshBuffer(0u)->getPipeline()->clone());
+
+    //get the simple geometry shader data and turn it into ICPUSpecializedShader
+    auto shaderData = device->getAssetManager()->getFileSystem()->loadBuiltinData("irr/builtin/shaders/testGeomShader.geom");
+    auto unspecializedShader = core::make_smart_refctd_ptr<asset::ICPUShader>(std::move(shaderData), asset::ICPUShader::buffer_contains_glsl);
+    auto shader = core::make_smart_refctd_ptr<asset::ICPUSpecializedShader>(std::move(unspecializedShader), asset::ISpecializedShader::SInfo({}, nullptr, "main", asset::ISpecializedShader::ESS_GEOMETRY));
+    pipeline_cp->setShaderAtIndex(asset::ICPURenderpassIndependentPipeline::ESSI_GEOMETRY_SHADER_IX, shader.get());
+
+    //replace the ICPURenderpassIndependentPipeline with a copy that uses the replaced geometry shader
+    mesh_raw->getMeshBuffer(0u)->setPipeline(std::move(pipeline_cp));
+
+    //create buffers for the geometry shader
+    auto lineCountBuffer = driver->createDeviceLocalGPUBufferOnDedMem(sizeof(uint32_t));
```

Comment:
this needs to be `roundUp(sizeof(DrawIndirectParameters_t),16)` and you need to fill it with sane values for everything except for the `count` which should be 0